odbc/src/api/data_tests.rs

```diff
@@ -2490,8 +2491,8 @@ mod unit {
                         month: 11,
                         day: 28,
                     },
-                    SqlReturn::SUCCESS,
-                    "",
+                    SqlReturn::SUCCESS_WITH_INFO,
+                    "[MongoDB][API] floating point data \"2014-11-28 12:00:09.0 +00:00:00\" was truncated to fixed point",
```

Comment:
Looks like our error message it a little bit too much tailored for numeric data. Do we want a date and time version of this one?

Reply:
I was wondering the same too; I wasn't sure if creating multiple error types for the same sql state would cause any downstream issues. However if it is just as easy as adding messages to the `ODBCError` enum and having them map to the `FRACTIONAL_TRUNCATION` sql state I agree!